Meadowlark-js/packages/meadowlark-authz-server/src/handler/VerifyToken.ts

```diff
@@ -0,0 +1,137 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+import querystring from 'node:querystring';
+import { Logger } from '@edfi/meadowlark-core';
+import type { AuthorizationResponse } from './AuthorizationResponse';
+import type { AuthorizationRequest } from './AuthorizationRequest';
+import type { VerifyTokenBody } from '../model/VerifyTokenBody';
+import { BodyValidation, validateVerifyTokenBody } from '../validation/BodyValidation';
+import { ensurePluginsLoaded } from '../plugin/AuthorizationPluginLoader';
+import { extractAuthorizationHeader } from './AuthorizationHeader';
+import {
+  hasAdminRole,
+  introspectBearerToken,
+  IntrospectionResponse,
+  validateTokenForAccess,
+  ValidateTokenResult,
+} from '../security/TokenValidator';
+
+const moduleName = 'VerifyToken';
+
+type ParsedVerifyTokenBody =
+  | { isValid: true; verifyTokenBody: VerifyTokenBody }
+  | { isValid: false; failureMessage: string };
+
+/**
+ * Parses the request which is form url-encoded with two tokens. One is the bearer token itself and the
+ * other is the token for verification/introspection.
+ *
+ * This function returns the request elements as an object, or a failure message if the request
+ * is not valid.
+ */
+function parseVerifyTokenBody(authorizationRequest: AuthorizationRequest): ParsedVerifyTokenBody {
+  if (authorizationRequest.body == null) return { isValid: false, failureMessage: 'Request body is empty' };
+
+  let unvalidatedBody: any;
+
+  // startsWith accounts for possibility of the content-type being with or without encoding
+  if (!authorizationRequest.headers['content-type']?.startsWith('application/x-www-form-urlencoded')) {
+    Logger.debug(
+      `${moduleName}.parseVerifyTokenBody: Requires application/x-www-form-urlencoded content type`,
+      authorizationRequest.traceId,
+    );
+    return { isValid: false, failureMessage: 'Malformed body' };
+  }
+
+  try {
+    unvalidatedBody = querystring.parse(authorizationRequest.body);
+  } catch (error) {
+    Logger.debug(`${moduleName}.parseRequestTokenBody: Malformed body`, authorizationRequest.traceId);
+    return { isValid: false, failureMessage: 'Malformed body' };
```

Comment:
Rather than having to rediscover this with an upcoming ticket on "malformed body", can you go ahead and report the error now? So long as we ensure the _content_ of the body is not part of the error message.

Reply:
Done